### PR TITLE
Remove boilerplate from scheduler test.

### PR DESCRIPTION
**Pull request by lszekeres**
Compare only the docker startup part of the script, as the rest doesn't contain
any template variables. Checking the fixed part of the template is not
interesting for the test and just adds unnecessary boilerplate.

### CODE REVIEW

File: experiment/test_scheduler.py [hunk 1]
```diff
@@ -201,7 +171,9 @@ def _test_create_trial_instance(benchmark, expected_image, expected_target,
         startup_script=expected_startup_script_path)
 
     with open(expected_startup_script_path) as file_handle:
-        assert file_handle.read() == expected_format_string.format(
+        content = file_handle.read()
+        script_for_docker = content[content.find('## Start docker.'):]
```
Comment (inferno-chromium) [inline experiment/test_scheduler.py:175]: maybe assert '## Start docker.' in content first.

File: experiment/resources/runner-startup-script-template.sh [hunk 2]
```diff
@@ -13,8 +13,16 @@
...
+## Configure the host.
+
+# Make everything ptrace-able.
 echo 0 > /proc/sys/kernel/yama/ptrace_scope
+
+# Do not notify external programs about core dumps.
 echo core >/proc/sys/kernel/core_pattern
+
+## Start docker.
+
```
Comment (inferno-chromium) [inline experiment/resources/runner-startup-script-template.sh:25]: can remove newline.

**Comment by lszekeres** [comment]
> Not sure I think this is an improvement.
> I think the boilerplate (ugly as it is) is nicer than having extra logic to avoid it, the code is easier to understand with the boilerplate.

The motivation is https://github.com/google/fuzzbench/pull/247 and the problem is redundancy: we have the same code in three different places, which means that if we want to make a change, we need to update it in three places. The part of the test that checks the template instantiation is useful, but the part that tests if X=X is less. wdty?

**Comment by jonathanmetzman** [comment]
> > Not sure I think this is an improvement.
> > I think the boilerplate (ugly as it is) is nicer than having extra logic to avoid it, the code is easier to understand with the boilerplate.
> 
> The motivation is #247 and the problem is redundancy: we have the same code in three different places, which means that if we want to make a change, we need to update it in three places. The part of the test that checks the template instantiation is useful, but the part that tests if X=X is less. wdty?

Heh, you know how to convince me, I think having to change tests in multiple places because they test the same thing is very bad. Yep overall this LGTM, lemme just review for real now.

File: experiment/test_scheduler.py (hunk 1 above)
Comment (lszekeres) [reply experiment/test_scheduler.py:175]: Good idea! done

File: experiment/resources/runner-startup-script-template.sh (hunk 2 above)
Comment (lszekeres) [reply experiment/resources/runner-startup-script-template.sh:25]: done

**Pull request merged**